### ISSUE TITLE
Maximising likelihood as objective function

### ISSUE THREAD

**Issue by kyujinson03** [keras-team/keras#14544]
\begin{align}
$\frac{1}{N} \sum_{n=1}^{N} log \sigma (f_{\theta}(x) - f_{\theta}(x'))$
\end{align}

$f_{\theta}(x)$ and $f_{\theta}(x')$ with weight $\theta$ are the outputs of a neural network, and they can be any scalar values. Using Keras, how can I specify a loss function that takes in 2 output values of the nn?
